src/main/java/seedu/coinflip/Coinflip.java

```diff
@@ -0,0 +1,329 @@
+    private void saveBalance(int balance) throws CoinflipFileException {
+        try {
+            FileWriter writer = new FileWriter(saveFilePath);
+            writer.write("Balance\n");
+            writer.write(balance + "\n");
+            writer.close();
+        } catch (IOException e) {
+            throw new CoinflipFileException(CoinflipFileException.SAVE_FILE_CANNOT_SAVE);
+        }
+    }
+
+    //@@author CRL006
+    private void saveToFile() throws CoinflipFileException {
+        saveBalance(balance);
+    }
+
+    //@@author timothyloh0523
+    private void check(String[] words) throws CoinflipException {
+        if (words.length != 2) {
+            throw new CoinflipException(CoinflipException.CHECK_INVALID_FORMAT);
+        }
+
+        if (!words[1].equals("balance") && !words[1].equals("bet")) {
+            throw new CoinflipException(CoinflipException.CHECK_INVALID_FORMAT);
+        }
+
+        if (words[1].equals("balance")) {
+            Printer.printBalance(balance);
+        } else {
+            Printer.printBetAmount(betAmount);
+        }
+    }
+
+    //@@author OliverQiL
+    private void change(String[] words) throws CoinflipException {
+        try {
+            betAmount = Integer.parseInt(words[1]);
+
+            if (betAmount < 0) {
+                throw new CoinflipException(CoinflipException.BET_AMOUNT_INVALID_FORMAT);
+            }
+
+            if (betAmount > balance) {
+                Printer.printNotEnoughCoins();
+                return;
+            }
+
+            Printer.printBetAmount(betAmount);
+
+        } catch (NumberFormatException | ArrayIndexOutOfBoundsException e) {
+            throw new CoinflipException(CoinflipException.BET_AMOUNT_INVALID_FORMAT);
+        }
+    }
+
+    //@@author wongyihao0506
+    private void bet(String[] words) throws CoinflipException {
+        if (words.length != 2) {
+            throw new CoinflipException(CoinflipException.FLIP_INVALID_FORMAT);
+        }
+
+        if (!words[1].equals("heads") && !words[1].equals("tails")) {
+            throw new CoinflipException(CoinflipException.FLIP_INVALID_FORMAT);
+        }
+
+        if (betAmount > balance) {
+            Printer.printNotEnoughCoins();
+            return;
+        }
+
+        Random random = new Random();
+        String coinFlip = random.nextBoolean() ? "Heads" : "Tails";
+        Boolean outcome = coinFlip.equalsIgnoreCase(words[1]);
+
+        if (outcome) {
+            balance += getBetAmount();
+        } else {
+            balance -= getBetAmount();
+        }
+
+        Printer.printBetAmount(betAmount);
+        Printer.printFlipOutcome(coinFlip, outcome, betAmount);
+        Printer.printBalance(balance);
+
+        assert balance >= 0 : "balance should be more than or equal to 0";
+    }
+
+    //@@author HTY2003
+    
+    /**
+     * Runs main Coinflip program, which waits for next line of user input
+     * before outputting an appropriate response
```

Comment:
nitpick, but following the sample javadoc guides, should send the comment with a fullstop
https://nus-cs2113-ay2425s2.github.io/website/se-book-adapted/chapters/documentation.html#javadoc